source/StartConditionsPanel.cpp

```diff
@@ -0,0 +1,292 @@
+/* StartConditionsPanel.cpp
+Copyright (c) 2020 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "StartConditionsPanel.h"
+
+#include "Color.h"
+#include "Command.h"
+#include "ConversationPanel.h"
+#include "DataFile.h"
+#include "Dialog.h"
+#include "Files.h"
+#include "FillShader.h"
+#include "GameData.h"
+#include "GameWindow.h"
+#include "Information.h"
+#include "Interface.h"
+#include "LoadPanel.h"
+#include "MainPanel.h"
+#include "Messages.h"
+#include "PlayerInfo.h"
+#include "Planet.h"
+#include "Preferences.h"
+#include "Rectangle.h"
+#include "ShipyardPanel.h"
+#include "StarField.h"
+#include "System.h"
+#include "text/alignment.hpp"
+#include "text/DisplayText.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "text/Format.h"
+#include "text/layout.hpp"
+#include "text/truncate.hpp"
+#include "UI.h"
+
+#include <SDL2/SDL.h>
+
+#include "gl_header.h"
+
+using namespace std;
+
+StartConditionsPanel::StartConditionsPanel(PlayerInfo &player, UI &gamePanels, LoadPanel *loadPanel)
+	: player(player), gamePanels(gamePanels), loadPanel(loadPanel)
+{
+	if(!GameData::Start().empty())
+	{
+		chosenStart = GameData::Start().back();
+		hasChosenStart = true;
+	}
+	const Interface *startConditionsMenu = GameData::Interfaces().Find("start conditions menu");
+	
+	if(startConditionsMenu)	
+	{
+		descriptionBox =   startConditionsMenu->GetBox("start description");
+		entryBox =         startConditionsMenu->GetBox("start entry");
+		entryListBox =     startConditionsMenu->GetBox("start entry list");
+		entryInternalBox = startConditionsMenu->GetBox("start entry internal");	
+	}
+	
+	size_t i = 0;
+	// Fill up the startConditionsClickZones vector
+	for(const StartConditions &it : GameData::Start())
+	{
+		// emplace_back will implicitly instantiate a ClickZone
+		startConditionsClickZones.emplace_back(
+			Rectangle::FromCorner(
+				Point(entryListBox.Left(), entryListBox.Top() + i * entryBox.Height()),
+				entryBox.Dimensions()
+			),
+			it);
+		i++;
```

Comment:
Inconsistent styling here ("flag" and postfix increment). There are other occurrences in this file that should also be addressed.

Reply:
What do you mean by "flag" and postfix increment? The postfix increment and the `i` variables are there to move the clickzones down